src/react/AstarteClient.js

```diff
@@ -494,3 +516,70 @@ function registerTrigger(channel, triggerPayload) {
       .receive("error", (err) => { reject(err) });
   });
 }
+
+function isValidRealmName(name) {
+  return RegExp('^[a-z][a-z0-9]{0,47}$').test(name) &&
+        !name.startsWith("astarte") &&
+        !name.startsWith("system");
+}
+
+function isTokenExpired(decodedTokenObject) {
+  if (decodedTokenObject.exp) {
+    const posix = Number.parseInt(decodedTokenObject.exp);
+    const expiry = new Date(posix * 1000);
+    const now = new Date();
+
+    return (expiry <= now);
+  } else {
+    return false;
+  }
+}
+
+function hasAstarteClaims(decodedTokenObject) {
+  // AppEngine API
+  if ("a_aea" in decodedTokenObject) {
+    return true;
+  }
+
+  // Realm Management API
+  if ("a_rma" in decodedTokenObject) {
+    return true;
+  }
+
+  // Pairing API
+  if ("a_pa" in decodedTokenObject) {
+    return true;
+  }
+
+  // Astarte Channels
+  if ("a_ch" in decodedTokenObject) {
+    return true;
+  }
+
+  return false;
+}
+
+function validateAstarteToken(token) {
+  const decoded = jwt.decode(token, {complete: true});
+
+  let status;
+
+  if (decoded) {
+    if (isTokenExpired(decoded.payload)) {
+      status = "expired";
+
+    } else if (!hasAstarteClaims(decoded.payload)) {
+      status = "notAnAstarteToken";
+
+    } else {
+      status = "valid";
+    }
+
+  } else {
+    status = "invalid";
+  }
+
+  return status;
+}
+
+export { AstarteClient, isValidRealmName, validateAstarteToken };
```

Comment:
How about leaving AstarteClient as the default export and add utility methods to the client? These two could be `static` methods

Reply:
I also thought about that, but looking at other libraries like react, jwt and uuid, they all seems to prefer importing function separately.

Reply:
Totally legit. One common thing I saw nevertheless is that often there is a default export which includes the main class / functionality. So this is actually a question about what style to maintain throughout the codebase